app/controllers/publish/providers/allocations_controller.rb

```diff
@@ -0,0 +1,26 @@
+module Publish
+  module Providers
+    class AllocationsController < PublishController
+      def index
+        authorize(Allocation)
+
+        previous_recruitment_cycle = recruitment_cycle.previous
```

Comment:
Feels like all this data preparation should go into smaller methods and just instantiate one instance variable for the view on line 20

Reply:
done